dataloader/dataloader.hpp

```diff
@@ -117,6 +121,112 @@ class DataLoader
+                                  const double validRatio = 0.2,
+                                  const bool shuffle = true,
+                                  const std::vector<std::string>& augmentation =
+                                      std::vector<std::string>(),
+                                  const double augmentationProbability = 0.2,
+                                  const bool absolutePath = false,
+                                  const std::string& baseXMLTag = "annotation",
+                                  const std::string& imageNameXMLTag =
+                                      "filename",
+                                  const std::string& sizeXMLTag = "size",
+                                  const std::string& objectXMLTag = "object",
+                                  const std::string& bndboxXMLTag = "bndbox",
+                                  const std::string& classNameXMLTag = "name",
+                                  const std::string& x1XMLTag = "xmin",
+                                  const std::string& y1XMLTag = "ymin",
+                                  const std::string& x2XMLTag = "xmax",
+                                  const std::string& y2XMLTag = "ymax");
+
+  /**
+   * Load all images from directory.
+   *
+   * @param imagesPath Path to all images.
+   * @param dataset Armadillo type where images will be loaded.
+   * @param labels Armadillo type where labels will be loaded.
+   * @param imageWidth Width of images in dataset.
+   * @param imageHeight Height of images in dataset.
+   * @param imageDepth Depth of images in dataset.
+   * @param label Label which will be assigned to image.
+   * @param augmentation Vector strings of augmentations supported by mlpack.
+   * @param augmentationProbability Probability of applying augmentation to a particular cell.
+   */
+  void LoadAllImagesFromDirectory(const std::string& imagesPath,
+                                  DatasetX& dataset,
+                                  DatasetY& labels,
+                                  const size_t imageWidth,
+                                  const size_t imageHeight,
+                                  const size_t imageDepth,
+                                  const size_t label = 0);
+
+  /**
+   * Load all images from directory.
+   *
+   * @param pathToDataset Path to all folders containing all images.
+   * @param imageWidth Width of images in dataset.
+   * @param imageHeight Height of images in dataset.
+   * @param imageDepth Depth of images in dataset.
+   * @param trainData Determines whether data is training set or test set.
+   * @param shuffle Boolean to determine whether or not to shuffle the data.
+   * @param validRatio Ratio of dataset to be used for validation set.
+   * @param augmentation Vector strings of augmentations supported by mlpack.
+   * @param augmentationProbability Probability of applying augmentation to a particular cell.
```

Comment:
do you mean particular cell or image?

Reply:
I used cell here because some augmentation maybe be applied to csv like addition of Gaussian noise. Kindly let me know if I need to change it.

Reply:
By cell you mean pixel? I am not sure about the terminology here

Reply:
I mean the whole image. Say, we have n images then we apply augmentation to augmentationProbability * n images. Let me change it to data point, does that make sense or any another word.

Reply:
IMO image would sound better. Because even if it is csv file the augmentation is still being applied to an image.

Reply:
Let me know what you think

Reply:
Done, in the last commit :)